search.py

```diff
@@ -1,67 +1,73 @@
-import requests
 import click
-import webbrowser
 import pandas
-import os
+import time
 from bs4 import BeautifulSoup
+from bs4.element import Tag
 from pathlib import Path
+from selenium import webdriver
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
 
 
 @click.command()
 @click.option('-s', '--search', "user_query", default='World', help="User's search query")
-@click.option('-b', '--browser', "browser", default='chrome', help="Browser to use")
+@click.option('-b', '--browser', "browser", default='Chrome', help="Browser to use")
 @click.option('-o', '--output', "output", default='not', help="File to save the results")
 def main(user_query, browser, output):
     print(f"Searching for {user_query} in {browser}...")
+    if browser.capitalize() == 'Firefox':
+        firefox_capabilities = DesiredCapabilities.FIREFOX
+        firefox_capabilities['marionette'] = False
+        driver = webdriver.Firefox(executable_path="./drivers/geckodriver")
+    elif browser.capitalize() == 'Edge':
+        driver = webdriver.Edge(executable_path="./drivers/msedgedriver")
+    else:
+        driver = webdriver.Chrome(executable_path="./drivers/chromedriver")
 
-    replace_whitespace = user_query.replace(' ', '+')
-    number_of_search_results = 10
-    url = f"https://www.google.com/search?q={replace_whitespace}&num={number_of_search_results}"
-    webbrowser.get(browser).open(url)
-    requests_result = requests.get(url)
-    soup_link = BeautifulSoup(requests_result.content, "html.parser")
-    links = soup_link.find_all("a")
+    driver.get("https://www.google.com")
 
-    my_file = Path(output)
+    search_bar = driver.find_element(by="name", value="q")
+    search_bar.clear()
+    search_bar.send_keys(user_query)
+    search_bar.send_keys(Keys.RETURN)
 
-    if my_file.exists():
-        to_file(links, my_file)
-    else:
-        to_console(links)
+    time.sleep(3)
 
+    soup = BeautifulSoup(driver.page_source, "html.parser")
+    result_div = soup.find_all('div', attrs={'class': 'g'})
 
-def to_file(links, my_file):
-    result = open(my_file, "w")
-    result.write("Title,URL\n")
+    links = []
+    titles = []
 
-    add_results(links, result)
+    for result in result_div:
+        link = result.find('a', href=True)
+        title = result.find('h3')
 
-    print("Done!")
+        if isinstance(title, Tag):
+            title = title.get_text()
 
+        if link != '' and link['href'][0] != "/" and title != '':
+            links.append(link['href'])
+            titles.append(title)
 
-def to_console(links):
-    result = open('result.csv', "w")
-    result.write("Title,URL\n")
+    my_file = Path(output)
 
-    add_results(links, result)
+    if my_file.exists():
```

Comment:
what would happen if the folder does not exist?

Reply:
If folder or file does not exist or if there a typo - result will be shown in terminal
![image](https://user-images.githubusercontent.com/15080221/167109619-22febcb3-9a3e-4618-95b0-2da59a72a541.png)